src/Modules/OrchardCore.Commerce/Services/CardPaymentService.cs

```diff
@@ -0,0 +1,157 @@
+        _logger = logger;
+    }
+
+    public async Task<PaymentIntent> CreatePaymentAsync(ConfirmPaymentRequest request)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        var totals = await currentShoppingCart.CalculateTotalsAsync(_priceService, _priceSelectionStrategy);
+
+        // Same here as on the checkout page: Later we have to figure out what to do if there are multiple
+        // totals i.e., multiple currencies.
+        var defaultTotal = totals.FirstOrDefault();
+
+        var paymentIntent = new PaymentIntent();
+        var requestOptions =
+            new RequestOptions
+            {
+                ApiKey = (await _siteService.GetSiteSettingsAsync())
+                    .As<StripeApiSettings>()
+                    .SecretKey
+                    .DecryptStripeApiKey(_dataProtectionProvider, _logger),
+            };
+
+        if (request.PaymentMethodId != null)
+        {
+            var paymentIntentOptions = new PaymentIntentCreateOptions
+            {
+                // We need to remove the decimal points and convert the value (decimal) to long.
+                // https://stripe.com/docs/currencies#zero-decimal
+                Amount = long
+                .Parse(
+                    string.Join(
+                        string.Empty,
+                        defaultTotal.ToString().Where(char.IsDigit)),
+                    CultureInfo.InvariantCulture),
+                Currency = defaultTotal.Currency.CurrencyIsoCode,
+                Description = "Orchard Commerce Test Stripe Card Payment",
+                ConfirmationMethod = "manual",
+                Confirm = true,
+                PaymentMethod = request.PaymentMethodId,
+
+                // If shipping is implemented, it needs to be added here too.
+                // Shipping =
+                // ReceiptEmail = viewModel.Email,
+            };
+
+            paymentIntent = _paymentIntentService.Create(paymentIntentOptions, requestOptions);
+        }
+
+        if (request.PaymentIntentId != null)
+        {
+            paymentIntent = _paymentIntentService.Confirm(
+                request.PaymentIntentId,
+                new PaymentIntentConfirmOptions(),
+                requestOptions);
+        }
+
+        return paymentIntent;
+    }
+
+    public async Task CreateOrderFromShoppingCartAsync(PaymentIntent paymentIntent)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        var totals = await currentShoppingCart.CalculateTotalsAsync(_priceService, _priceSelectionStrategy);
+        var defaultTotal = totals.FirstOrDefault();
+
+        var order = await _contentManager.NewAsync("Order");
+        var orderId = Guid.NewGuid();
+
+        order.DisplayText = "Order " + orderId;
+
+        // To-do when other parts of the checkout is implemented (notes).
+        // order.Alter<HtmlBodyPart>(htmlBodyPart => htmlBodyPart.
```

Comment:
Same, mention it in the [checkout issue](https://github.com/OrchardCMS/OrchardCore.Commerce/issues/16) and include the comment URL here.